### PR TITLE
XML to TOON Conversion in JToon

### PR DESCRIPTION
**Pull request by badpirogrammer2**
1. Core Implementation Complete: Full functionality for converting XML documents to TOON format is implemented.

2. New API Methods: Extended the main JToon.java class with new static methods for encoding:
    
    encodeXml(String xml) (Basic conversion)
    
    encodeXml(String xml, EncodeOptions options) (Conversion with custom options)

3. Dedicated Normalizer: Created XmlNormalizer.java to handle XML parsing using Jackson's XmlMapper and convert it to an intermediate JsonNode object.

4. Dependency Integration: Integrated the jackson-dataformat-xml dependency for robust XML parsing capabilities.

5. Comprehensive Structure Support: The conversion successfully handles:

    Simple and nested XML elements (converted to TOON objects).
    
    XML attributes (included as additional TOON fields).
    
    XML arrays (converted to TOON arrays).
    
    Preserves element hierarchy and ordering.

6. Full Options Support: The new methods support existing EncodeOptions for:

    Custom indentation levels.
    
    Different delimiters (comma, pipe, tab).
    
    Length markers for arrays.

7. Robust Error Handling: Implemented checks to throw an IllegalArgumentException for:

    Null or empty XML strings.
    
    Malformed XML documents.
    
    Invalid XML structure.

8. Extensive Testing:

    449 total tests are passing (BUILD SUCCESSFUL).
    
    Tests cover positive scenarios, negative/error handling, and complex edge cases (nested structures, attributes).

9. Documentation: Added comprehensive JavaDoc documentation with clear signatures and usage examples.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -13,13 +13,16 @@ apply from: 'gradle/publishing.gradle'
 apply from: 'gradle/verification.gradle'
 
 java {
-    // Only enforce toolchain in CI to ensure consistent Java 17 builds
```
Comment (felipestanzani) [inline build.gradle]: This change is disallowed. The project was downgraded to Java 17, allowing more developers to use it..

File: src/test/java/com/felipestanzani/jtoon/JToonTest.java
Comment (felipestanzani) [inline src/test/java/com/felipestanzani/jtoon/JToonTest.java]: You can't change the current implemented test names this way. Separate your tests inside a new Nested class.

File: DebugXml.java
Comment (felipestanzani) [inline DebugXml.java]: This project follows standards. These loose files are not allowed.

File: TestXml.java
Comment (felipestanzani) [inline TestXml.java]: This project follows standards. These loose files are not allowed.

Why a Test here if we have a Test Project?

**Comment by badpirogrammer2** [comment]
@felipestanzani  please let me know what is missing exept the readme?
